packages/riipen-ui/src/components/Link.test.jsx

```diff
@@ -0,0 +1,178 @@
+    it("renders given children", () => {
+      const child = <span>link</span>;
+
+      const wrapper = mount(<Link href="google.com">{child}</Link>);
+
+      expect(wrapper.childAt(0).contains(child)).toEqual(true);
+    });
+  });
+
+  describe("classes prop", () => {
+    it("sets custom class", () => {
+      const classes = ["classOne"];
+
+      const wrapper = mount(
+        <Link classes={classes} href="google.com">
+          <span>link</span>
+        </Link>
+      );
+
+      expect(
+        wrapper
+          .find("Link")
+          .childAt(0)
+          .hasClass(classes[0])
+      ).toEqual(true);
+    });
+  });
+
+  describe("color prop", () => {
+    it("applies valid custom color class", () => {
+      const color = "negative";
+
+      const wrapper = mount(
+        <Link color={color} href="google.com">
+          <span>link</span>
+        </Link>
+      );
+
+      expect(
+        wrapper
+          .find("Link")
+          .childAt(0)
+          .hasClass(color)
+      ).toEqual(true);
+    });
+
+    it("gives an error when given an invalid color", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+
+      mount(
+        <Link color="pink" href="google.com">
+          <span>link</span>
+        </Link>
+      );
+
+      expect(errors).toHaveBeenCalled();
+    });
+  });
+
+  describe("component prop", () => {
+    it("sets custom component", () => {
+      const component = "div";
+
+      const wrapper = mount(
+        <Link component={component} href="google.com">
+          <span>link</span>
+        </Link>
+      );
+
+      expect(
+        wrapper
+          .find("Link")
+          .childAt(0)
+          .name()
+      ).toEqual(component);
+    });
+
+    it("gives an error when given an invalid component", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const component = "abcd";
+
+      mount(
+        <Link component={component} href="google.com">
+          <span>link</span>
+        </Link>
+      );
+
+      expect(errors).toHaveBeenCalled();
+    });
+  });
+
+  describe("focusVisible state", () => {
+    it("applies focusVisible class when focus event occurs", () => {
+      const wrapper = mount(
+        <Link href="google.com">
+          <span>link</span>
+        </Link>
+      );
+
+      wrapper.simulate("focus");
+
+      expect(
+        wrapper
+          .find("Link")
+          .childAt(0)
+          .hasClass("focusVisible")
+      ).toEqual(true);
+    });
+
+    it("does not apply focusVisible class when blur event occurs", () => {
+      const wrapper = mount(
+        <Link href="google.com">
+          <span>link</span>
+        </Link>
+      );
+
+      wrapper.simulate("focus");
+      wrapper.simulate("blur");
+
+      expect(
+        wrapper
+          .find("Link")
+          .childAt(0)
+          .hasClass("focusVisible")
+      ).toEqual(false);
+    });
+  });
+});
```

Comment:
Should there be tests for the underline prop? It looks like we could move the needed info to inline style arguments.

Reply:
underline prop is not testable :(